cardano-api/src/Cardano/Api/Fees.hs

```diff
@@ -1025,17 +1026,18 @@ makeTransactionBodyAutoBalance eraInMode systemstart history pparams
     -- fit within the encoding size we picked above when calculating the fee.
     -- Yes this could be an over-estimate by a few bytes if the fee or change
     -- would fit within 2^16-1. That's a possible optimisation.
-    txbody3 <-
-      first TxBodyError $ -- TODO: impossible to fail now
-        createAndValidateTransactionBody txbodycontent1 {
+    let finalTxBodyContent = txbodycontent1 {
           txFee  = TxFeeExplicit explicitTxFees fee,
           txOuts = accountForNoChange
                      (TxOut changeaddr balance TxOutDatumNone ReferenceScriptNone)
                      (txOuts txbodycontent),
           txReturnCollateral = retColl,
           txTotalCollateral = reqCol
         }
-    return (BalancedTxBody txbody3 (TxOut changeaddr balance TxOutDatumNone ReferenceScriptNone) fee)
+    txbody3 <-
+      first TxBodyError $ -- TODO: impossible to fail now
```

Comment:
I think the TODO comment needs elaboration.  The TODO should say what needs to be done.

Reply:
That was there previously but I'll update it

Reply:
Thanks!